src/index.js

```diff
@@ -1,5 +1,185 @@
 const OPTIMIZELY_CRIER_SEMANTIC_VERSION = '%%PACKAGE_VERSION%%';
 
+/**
+ * Optimizely Segment Custom Destination Function:
+ *  https://github.com/circAssimilate/segment-optimizely-crier
+ *
+ * Learn more about destination functions API:
+ *  https://segment.com/docs/connections/destinations/destination-functions
+ */
+
+/**
+ * @description https://segment.com/docs/connections/spec/track/#example
+ * @param {SegmentTrackEvent} segmentEvent
+ * {
+ *   anonymousId: '61db7c12-c437-4efe-9123-f3e2031903b5',
+ *   event: 'Change History Page View',
+ *   integrations: {
+ *     Optimizely: {
+ *       attributes: {
+ *         account_id: '5935064',
+ *         created_epoch_date: 1301345505680,
+ *         total_revenue_three_months: 0,
+ *         is_internal_user: true,
+ *       },
+ *       userId: '5935064',
+ *     },
+ *   },
+ *   messageId: 'ajs-0c224f035fce1c9adc62d6c06035439e',
+ *   timestamp: '2020-03-09T21:33:11.594Z',
+ *   type: 'track',
+ *   userId: '5935064_4df8049ed4b011e3aa46635c25628756',
+ * }
+ * @param {FunctionSettings} settings
+ */
+async function onTrack({ event, integrations, timestamp }, { apiKey }) {
+  const { attributes, userId } =
+    (integrations ? integrations : {}).Optimizely || {};
+
+  if (!userId) {
+    throw new InvalidEventPayload(
+      '[OPTIMIZELY] Optimizely.integrations.userId not present'
+    );
+  }
+
+  console.log(`[OPTIMIZELY] Optimizely.integrations.userId "${userId}"`); // eslint-disable-line no-console
+
+  let { accountId, attributeKeyToIdMap, eventKeyToIdMap } =
+    typeof OPTIMIZELY_DATA !== 'undefined' ? OPTIMIZELY_DATA : {};
```

Comment:
See comment above in the gulpfile. I think it makes more sense to just inject the SDK key and then download the datafile at runtime.

This doesnt need to be super performant, and there is likely caching going on automatically in whatever node environment segment is using to run this.